### ISSUE TITLE
Shooting an MMI in your hand shoots yourself

### ISSUE THREAD

**Issue by D3athrow-Issues** [d3athrow/vgstation13#9970]
(WEB REPORT BY: zealotron REMOTE: 172.93.109.202:7777)

> Revision (Should be above if you're viewing this from ingame!)
> c2f1c4b483ca3490acd0a6b6ef1a865f8652da96 
> 
> General description of the issue
> If you try to click on an MMI thats in your hand, with like a laser or possibly other guns, it says 

 Sawbones fires the laser gun point blank at Han Hunt!

but you shoot yourself

> What you expected to happen
> to shoot the mmi
> 
> What actually happened
> i shot myself
> 
> Steps to reproduce if possible
> get an mmi in one another
> the captains laser in another
> click the mmi

**Comment by Shadowmech88** [comment]
Are MMIs objects or mobs?
What happens if you click an object in your hand with a gun?
What happens if you pick up a small animal like a clown goblin and try to shoot it in your hand?

**Comment by Exxion** [comment]
No matter what, something in your hand is an item
Mobs you can hold are placed within holder items then placed in your hand
So what you are clicking is definitely an item

**Comment by Shadowmech88** [comment]
Odd that you'd be able to fire a gun at it at all, then. To my knowledge harm-intent-point-blank shooting doesn't work on objects.

**Comment by Exxion** [comment]
Yes, it is odd
Perhaps attacking the item calls attack on the mob within

**Comment by TheFriggenSawbones** [comment]
normally you can click on mmis with a shard or whatever and spam their chat log, so maybe it's like creating a laser underneath you and that's what doing it? idk man good luck coder friends

**Comment by Exxion** [comment]
Yep, attacking an MMI with an item attacks the contained mob if that item doesn't have a specific effect on the MMI
Which means @TheFriggenSawbones' guess is probably right on the money

**Comment by clusterfack** [comment]
MMIs have fucked up attack code so that you can 'attack' the brains inside of them. It's just shitty code and can be replaced.

**Comment by Killette2** [comment]
I miss being able to screwdriver eye stab MMIs, that got removed forever ago.

**Comment by ComicIronic** [comment]
From my memory, this is because bullets start at get_turf() and target get_turf() and you get hit in the Bump() or whatever.

Point being, drop something to shoot it.

**Comment by Killette2** [comment]
New wizard artifact when? It's a belt with the letter M, when you shoot someone with it it miniaturizes them and you're able to pick them up, throw them, ect, but they get some buffs like hiding under tables and dodging projectiles a lot easier.

To fix them back to normal size, just flip the M upside down to change it from miniaturize to wumbo.

**Comment by ComicIronic** [comment]
We do not have nearly the systems required to implement something like that sanely.

**Comment by Killette2** [comment]
We absolutely do not. There would be tons of bugs and issues.
